Master branch doesn't compile in MSVC 2017

### Godot version

fcbc50ec144df458aee75db94cdbf6396bd408ed

### System information

MSVC 2017

### Issue description

Seems like MSVC doesn't respect `using namespace` in parameters inside function definition.
```
drivers\gles3\storage\texture_storage.cpp(834): error C2872: 'Texture': ambiguous symbol
C:\Docs\Programs\godotengine\scene/resources/texture.h(46): note: could be 'Texture'
c:\docs\programs\godotengine\drivers\gles3\storage\texture_storage.h(145): note: or       'GLES3::Texture'
drivers\gles3\storage\texture_storage.cpp(834): error C2039: 'Type': is not a member of 'Texture'
C:\Docs\Programs\godotengine\scene/resources/texture.h(46): note: see declaration of 'Texture'
drivers\gles3\storage\texture_storage.cpp(834): error C2065: 'Type': undeclared identifier
drivers\gles3\storage\texture_storage.cpp(834): error C2146: syntax error: missing ')' before identifier 'p_type'
drivers\gles3\storage\texture_storage.cpp(834): error C2143: syntax error: missing ';' before '{'
drivers\gles3\storage\texture_storage.cpp(834): error C2447: '{': missing function header (old-style formal list?)
scons: *** [drivers\gles3\storage\texture_storage.windows.editor.dev.x86_64.obj] Error 2
```

Caused by 23603e409c04fd0238a45552ad7b07a175b528c6 (no PR)

Fix:
```diff
diff --git a/drivers/gles3/storage/texture_storage.cpp b/drivers/gles3/storage/texture_storage.cpp
index 6ccd4246e6..4d7e08c849 100644
--- a/drivers/gles3/storage/texture_storage.cpp
+++ b/drivers/gles3/storage/texture_storage.cpp
@@ -831,7 +831,7 @@ void TextureStorage::texture_proxy_initialize(RID p_texture, RID p_base) {
 	texture_owner.initialize_rid(p_texture, proxy_tex);
 }
 
-RID TextureStorage::texture_create_external(Texture::Type p_type, Image::Format p_format, unsigned int p_image, int p_width, int p_height, int p_depth, int p_layers, RS::TextureLayeredType p_layered_type) {
+RID TextureStorage::texture_create_external(GLES3::Texture::Type p_type, Image::Format p_format, unsigned int p_image, int p_width, int p_height, int p_depth, int p_layers, RS::TextureLayeredType p_layered_type) {
 	Texture texture;
 	texture.active = true;
 	texture.is_external = true;
```

### Steps to reproduce

Try to build

### Minimal reproduction project

N/A

--- a/drivers/gles3/storage/texture_storage.cpp
+++ b/drivers/gles3/storage/texture_storage.cpp
@@ -831,7 +831,7 @@ void TextureStorage::texture_proxy_initialize(RID p_texture, RID p_base) {
 	texture_owner.initialize_rid(p_texture, proxy_tex);
 }
 
-RID TextureStorage::texture_create_external(Texture::Type p_type, Image::Format p_format, unsigned int p_image, int p_width, int p_height, int p_depth, int p_layers, RS::TextureLayeredType p_layered_type) {
+RID TextureStorage::texture_create_external(GLES3::Texture::Type p_type, Image::Format p_format, unsigned int p_image, int p_width, int p_height, int p_depth, int p_layers, RS::TextureLayeredType p_layered_type) {
 	Texture texture;
 	texture.active = true;
 	texture.is_external = true;

